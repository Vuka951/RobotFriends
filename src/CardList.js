import React, { Component } from 'react'
import Card from './Card';

class CardList extends Component {
  render() {
    const {robots} = this.props;
    return (
      <div>
        {robots.map((user) => {
        return <Card name={user.name} email={user.email} id={user.id} key={user.id}/>
      })}
      </div>
    )
  }
}

export default CardList;