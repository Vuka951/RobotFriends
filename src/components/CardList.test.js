import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

it('expect to render card component', () => {
    const mockRobots = [{
        id: 1,
        name: 'Jimmy Jimmy',
        username: 'SmallJim',
        email: 'jimmy@gimmy.com'

    }]
    expect(shallow(<CardList robots={mockRobots}/>)).toMatchSnapshot();
})
