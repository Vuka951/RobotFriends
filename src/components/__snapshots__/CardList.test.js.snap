// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
      Array [
        Object {
          "email": "jimmy@gimmy.com",
          "id": 1,
          "name": "Jimmy Jimmy",
          "username": "SmallJim",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="jimmy@gimmy.com"
          id={1}
          name="Jimmy Jimmy"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "email": "jimmy@gimmy.com",
          "id": 1,
          "name": "Jimmy Jimmy",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="jimmy@gimmy.com"
            id={1}
            name="Jimmy Jimmy"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "email": "jimmy@gimmy.com",
            "id": 1,
            "name": "Jimmy Jimmy",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
